src/NSwag.Generation/Processors/OperationSummaryAndDescriptionProcessor.cs

```diff
@@ -22,29 +23,56 @@ public class OperationSummaryAndDescriptionProcessor : IOperationProcessor
         /// <returns>true if the operation should be added to the Swagger specification.</returns>
         public bool Process(OperationProcessorContext context)
         {
-            dynamic descriptionAttribute = context.MethodInfo.GetCustomAttributes()
-                .SingleOrDefault(a => a.GetType().Name == "DescriptionAttribute");
+            var attributes = context.MethodInfo.GetCustomAttributes().ToList();
 
-            if (descriptionAttribute != null)
+            ProcessSummary(context, attributes);
+            ProcessDescription(context, attributes);
+
+            return true;
+        }
+
+        private void ProcessSummary(OperationProcessorContext context, List<Attribute> attributes)
+        {
+            dynamic openApiOperationAttribute = attributes
+                .SingleOrDefault(a => a.GetType().Name == "OpenApiOperationAttribute");
+
+            string summary = openApiOperationAttribute?.Summary;
+
+            if (string.IsNullOrEmpty(summary))
             {
-                context.OperationDescription.Operation.Summary = descriptionAttribute.Description;
+                dynamic descriptionAttribute = attributes
+                    .SingleOrDefault(a => a.GetType().Name == "DescriptionAttribute");
+
+                summary = descriptionAttribute?.Description;
             }
-            else
+
+            if (string.IsNullOrEmpty(summary))
             {
-                var summary = context.MethodInfo.GetXmlDocsSummary();
-                if (summary != string.Empty)
-                {
-                    context.OperationDescription.Operation.Summary = summary;
-                }
+                summary = context.MethodInfo.GetXmlDocsSummary();
             }
 
-            var remarks = context.MethodInfo.GetXmlDocsRemarks();
-            if (remarks != string.Empty)
+            if (!string.IsNullOrEmpty(summary))
             {
-                context.OperationDescription.Operation.Description = remarks;
+                context.OperationDescription.Operation.Summary = summary;
             }
+        }
 
-            return true;
+        private void ProcessDescription(OperationProcessorContext context, List<Attribute> attributes)
+        {
+            dynamic openApiOperationAttribute = attributes
+                .SingleOrDefault(a => a.GetType().Name == "OpenApiOperationAttribute");
+
+            string description = openApiOperationAttribute?.Description;
+
+            if (string.IsNullOrEmpty(description))
+            {
+                description = context.MethodInfo.GetXmlDocsRemarks();
```

Comment:
Isnt this already done somewhere else?

Reply:
Hmmm.... Not quite sure if I understood your question correctly. Can you "expand" you idea - what is possibly wrong in the code snippet above?

Reply:
I think there is already a place where GetXmlDocsRemarks is used and set, or is this wrong? Isnt this somehow duplicated code?

Reply:
As far as I can see - OperationSummaryAndDescriptionProcessor class is the only place where GetXmlDocsRemarks is used. Source code before my suggested change has this call at line 41 in Process method, but i've split it to separate ProcessSummary and ProcessDescription methods.

Reply:
Ok then it's fine I think.